File: dziennik-cakephp/app/Model/TimeTable.php
<?php
class TimeTable extends AppModel {
  public $name = 'TimeTable';
  public $belongsTo = array('SchoolClass' => array('foreignKey' => 'class_id'));
  public $hasMany = array('Lesson' => array('dependent' => true));
  public $actsAs = array('Containable'); 

  function beforeValidate() {
    App::import('CakeSession', 'AuthComponent');
    $this->data['TimeTable']['class_id'] = CakeSession::read('Auth.User.class_id');
  }

  function findAllAvailableDaysByClassId($class_id) {
    $week = array(
      1 => 'Poniedziałek',
      2 => 'Wtorek',
      3 => 'Środa',
      4 => 'Czwartek',
      5 => 'Piątek',
      6 => 'Sobota',
      7 => 'Niedziela'
    );
    $days = $this->findAllByClassId($class_id);
    foreach($days as $day){
      unset($week[$day['TimeTable']['week_day']]);
    }
    return $week;
  }

  function findSelectableSubjects($class_id) {
    $_subjects = $this->SchoolClass->Subject->findAllByClassId($class_id);
    foreach ($_subjects as $value) {
      $subjects[$value['Subject']['id']] = $value['Subject']['name'];
    }
    return $subjects;
  }
}

File: app/models/time_table.rb
# coding: utf-8

class TimeTable < ActiveRecord::Base

  default_scope :order => 'week_day ASC'

  belongs_to :school_class

  has_many :lessons, :dependent => :destroy

  attr_accessible :week_day, :lessons_attributes

  validates :week_day, :presence => true

  accepts_nested_attributes_for :lessons, :reject_if => lambda { |lesson| lesson[:subject_id].blank? }, :allow_destroy => true

  def week_day_name
    case week_day
    when 0
      'Niedziela'
    when 1
      'Poniedziałek'
    when 2
      'Wtorek'
    when 3
      'Środa'
    when 4
      'Czwartek'
    when 5
      'Piątek'
    when 6
      'Sobota'
    end
  end

  private

end
